### PR TITLE
Industry workflow

### PR DESCRIPTION
**Pull request by hazemakhalek**
## Changes proposed in this Pull Request
This PR adds a new industry demand workflow that works on the basis of the UNSD energy balances as well as the industrial database

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: config.default.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ demand_data:
   update_data: true # if true, the workflow downloads the energy balances data saved in data/demand/unsd/data again. Turn on for the first run.
   base_year: 2019
 
+  other_industries: false # Whether or not to include industries that are not specified. some countries have has exageratted numbers, check carefully.
```
Comment (energyLS) [inline config.default.yaml:46]: Check where, can you give a hint for the user?

File: data/AL_production.csv [hunk 2]
```diff
@@ -0,0 +1,34 @@
+country,production[ktons/a],Year,source
```
Comment (energyLS) [inline data/AL_production.csv:1]: What happens with countries not on the list, is there any fallback or do we expect an error then?

File: scripts/build_industry_demand.py [hunk 3]
```diff
@@ -84,8 +102,151 @@ def country_to_nodal(industrial_production, keys):
...
+        "mining and quarrying",
+        "food and tobacco",
+        "paper pulp and print",
+        "wood and wood products",
+        "textile and leather",
+        "construction",
+        "other",
+    ]
+
+    emission_factors = {
```
Comment (energyLS) [inline scripts/build_industry_demand.py:121]: Can you provide a source here? Optionally move the dict to the config

File: scripts/prepare_sector_network.py [hunk 4]
```diff
@@ -2428,9 +2428,11 @@ def add_rail_transport(n, costs):
     # TODO follow the same structure as land transport and heat
 
     # Load industry demand data
-    industrial_demand = pd.read_csv(snakemake.input.industrial_demand)  # * 1e6
+    industrial_demand = pd.read_csv(
+        snakemake.input.industrial_demand, index_col=0, header=0
+    )  # * 1e6
 
-    industrial_demand.set_index("TWh/a (MtCO2/a)", inplace=True)
+    # industrial_demand.set_index("TWh/a (MtCO2/a)", inplace=True)
```
Comment (energyLS) [inline scripts/prepare_sector_network.py:2435]: Delete instead of commenting out

**Review by energyLS — CHANGES_REQUESTED** [review]

**Comment by energyLS** [comment]
@hazemakhalek great work! Some minor comments remaining, once they are clarified we are good to go.

File: data/AL_production.csv (hunk 2 above)
Comment (hazemakhalek) [reply data/AL_production.csv:1]: So far there will be an error. A fallback would be zero with a message stating that.

**Comment by hazemakhalek** [comment]
> @hazemakhalek great work! Some minor comments remaining, once they are clarified we are good to go.

Should be good to go. Waiting for the CI

**Review by energyLS — APPROVED** [review]

**Pull request merged**